From d1c82026d4d2c677fa22e157d973417f0c35b79d Mon Sep 17 00:00:00 2001
From: Alberto Milone <alberto.milone@canonical.com>
Date: Fri, 10 Jul 2015 16:52:29 +0200
Subject: [PATCH 1/2] Add support for Linux 4.1

Drop the IRQF_DISABLED flag, as it was removed in Linux 4.1

Note: this was also backported to 3.19.0-26 in Ubuntu

Signed-off-by: Alberto Milone <alberto.milone@canonical.com>
---
 firegl_public.c | 4 +++-
 kcl_acpi.c      | 7 +++++++
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/firegl_public.c b/firegl_public.c
index a8961ad..1a39fb7 100755
--- a/firegl_public.c
+++ b/firegl_public.c
@@ -3495,10 +3495,12 @@ int ATI_API_CALL KCL_InstallInterruptHandler(
         KCL_PUB_InterruptHandlerWrap,
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22)
         ((useMSI) ? (SA_INTERRUPT) : (SA_SHIRQ)),
-#else
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(4,1,0)
         //when MSI enabled. keep irq disabled when calling the action handler,
         //exclude this IRQ from irq balancing (only on one CPU) 
         ((useMSI) ? (IRQF_DISABLED) : (IRQF_SHARED)),    
+#else
+        ((useMSI) ? (0x0) : (IRQF_SHARED)),
 #endif
         dev_name,
         context);
diff --git a/kcl_acpi.c b/kcl_acpi.c
index 3a80984..aed08d8 100755
--- a/kcl_acpi.c
+++ b/kcl_acpi.c
@@ -861,7 +861,14 @@ void ATI_API_CALL KCL_ACPI_No_Hotplug(void* dev)
 #elif LINUX_VERSION_CODE >= KERNEL_VERSION(3,17,0)
     if(pdev)
     {
+#ifndef UTS_UBUNTU_RELEASE_ABI
+#define UTS_UBUNTU_RELEASE_ABI 0
+#endif
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,1,3) || (UTS_UBUNTU_RELEASE_ABI < 26 && LINUX_VERSION_CODE <= KERNEL_VERSION(3,19,8))
        pci_ignore_hotplug(pdev);
+#else
+       pdev->ignore_hotplug = 1;
+#endif
     }
 #endif
 }
-- 
1.9.1

